veracruz-utils/src/platform/nitro/nitro.rs

```diff
@@ -12,33 +12,29 @@
 
 use serde::{Deserialize, Serialize};
 
-/// The Status value returned by the Nitro enclave for operations
-/// This is intended to be received as a bincode serialized
-/// `NitroRootEnclaveMessage::Status`
-#[derive(Serialize, Deserialize, Debug)]
-pub enum NitroStatus {
-    /// The operation generating the message succeeded
-    Success,
-    /// The operation generating the message failed
-    Fail,
-    /// The requested operation is not yet implemented
-    Unimplemented,
-}
+use super::vm::{VeracruzSocketError, VMStatus};
 
-/// An enumerated type describing messages passed between the Runtime Manager
-/// and the Nitro Root Enclave
+use byteorder::{ByteOrder, LittleEndian};
+use err_derive::Error;
+use nix::errno::Errno::EINTR;
+use nix::sys::socket::{recv, send, MsgFlags};
+use std::os::unix::io::RawFd;
+
+/// An enumerated type describing messages passed between to/from the Runtime
+/// Manager enclave (These originate from the Untrusted Pass-through (Veracruz
+/// server)
 /// These messages are inteded to be serialized using bincode before transport,
 /// and deserialized using bincode after transport
 #[derive(Serialize, Deserialize, Debug)]
 pub enum NitroRootEnclaveMessage {
     /// A message generated by an operation that did not return data, but did
     /// return a status.
     /// Most operations return data, but if they fail, they will return a
-    /// status set to `NitroStatus::Fail` (or `NitroStatus::Unimplemented` if
-    /// it is not implemented).
+    /// status set to `VMStatus::Fail` (or `VMStatus::Unimplemented` if
+    /// it is not implmeneted).
     /// Parameters:
-    /// NitroStatus - the Status
-    Status(NitroStatus),
+    /// VMStatus - the Status
+    Status(VMStatus),
```

Comment:
Have you tested these Nitro changes? Remember that Nitro builds are not part of CI yet

Reply:
No, you're right this needs checking for Nitro!